### DISCUSSION TITLE
wanted to know will grule support nested fact structure

### DISCUSSION THREAD

**Discussion by nagarajtrivedi** [hyperjumptech/grule-rule-engine#205] [General]
Hi all, I wanted to know will grule support nested fact structure (a fact structure within a fact structure).

Thanks
Nagaraj

**Comment by newm4n** [comment]
You mean something like this ?

https://github.com/hyperjumptech/grule-rule-engine/blob/master/examples/Issue4_test.go

**Reply by newm4n** [reply]
Or this ?

https://github.com/hyperjumptech/grule-rule-engine/blob/1e65eaf14534cf78bafad43e2d38e19a4db0cfde/examples/JSONDataContext_test.go#L131

**Reply by newm4n** [reply]
Or even this ?
https://github.com/hyperjumptech/grule-rule-engine/blob/master/examples/FunctionCallChaining_test.go

**Reply by nagarajtrivedi** [reply]
Hi, all of them are really helpful to me.
Thanks for your quick response.

Thanks
Nagaraj
